### PR TITLE
[WIP] Prepare for CmfMenuBundle 2.0

### PR DESCRIPTION
**Pull request by WouterJ**

### CODE REVIEW

**Comment by WouterJ** [comment]
As KnpMenuBundle 2.1.0 is now released, I've updated this PR to fully work with the CmfMenuBundle 2.0. The only problem is a bug in PHPCR related packages, preventing the fixtures to be loaded.

File: src/AppBundle/Twig/MenuExtension.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public function __construct(Helper $helper, MatcherInterface $matcher)
+    {
+        $this->helper = $helper;
+        $this->matcher = $matcher;
+    }
+
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('app_get_current_menu_item', array($this, 'getCurrent')),
```
Comment (WouterJ) [inline src/AppBundle/Twig/MenuExtension.php]: I was unsure if this method is really usefull for other applications (it, at last, uses a not so nice implementation), so I decided to just put it in the AppBundle.

Comment (dbu) [inline src/AppBundle/Twig/MenuExtension.php]: so the knp menu no longer allows this?

**Comment by dbu** [comment]
we should drop php 5.3 and php 5.4 support from this repo.

**Comment by dbu** [comment]
can you please link to the phpcr-odm bug so we know when to look at this again?

File: src/AppBundle/Twig/MenuExtension.php (hunk 1 above)
Comment (WouterJ) [inline src/AppBundle/Twig/MenuExtension.php]: There is no getCurrent method in the NodeInterface anymore, as determining the current menu item is no tackled with matchers and voters.

**Comment by WouterJ** [comment]
@dbu https://github.com/doctrine/DoctrinePHPCRBundle/issues/219 and https://github.com/symfony-cmf/standard-edition/pull/43#issuecomment-144434813 are the related problems

File: src/AppBundle/Twig/MenuExtension.php (hunk 1 above)
Comment (dbu) [inline src/AppBundle/Twig/MenuExtension.php]: should we have a service that stores this decision and offers a getCurrentMenuItem method?

**Comment by lsmith77** [comment]
can you update the composer.lock now that DocumentManager::transactional() has been removed?

**Comment by WouterJ** [comment]
These changes made it in master via #299

**Pull request closed without merging**